packages/context/src/__tests__/unit/binding-sorter.unit.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Binding, sortBindingsByGroup} from '../..';
+
+describe('BindingSorter', () => {
+  const orderedGroups = ['log', 'auth'];
+  const groupTagName = 'group';
+  let bindings: Binding<unknown>[];
+  let sortedBindingKeys: string[];
+
+  beforeEach(givenBindings);
+  beforeEach(sortBindings);
+
+  it('sorts by group', () => {
+    assertOrder('logger1', 'logger2', 'auth1', 'auth2');
+  });
+
+  it('sorts by group - unknown group comes before known ones', () => {
+    assertOrder('metrics', 'logger1');
```

Comment:
a user experience nitpick:

My intuition is the unknown ones should come after known ones and sorted by alphabet by default.

Reply:
For global interceptors, unknown groups should come first as they expect less knowledge from other ones.

Reply:
If you want to change the behavior, you can have a custom sorter anyway.

Reply:
I see, fair enough.